proposals/module-linking/Binary.md

```diff
@@ -0,0 +1,238 @@
+# Module Linking Binary Format
+
+This document is intended to be a reference for the current state of the binary
+encoding of the module linking proposal. This is kept in sync with [the
+explainer](Explainer.md) and that is also recommended reading to understand this
+document. Currently this isn't striving to have a spec-level of detail, but
+hopefully it should be enough to get implementations off the ground and agreeing
+with each other!
+
+## Module-level updates
+
+At a high-level two new index spaces are added: modules and instances. Index
+spaces are also updated to be built incrementally as they are defined.
+
+Four new sections are added
+
+* Module section, `modulesec`, declares nested modules whose code comes
+  later. Only types are mentioned in this section.
+* Module Code section, `modulecodesec`, defines the nested modules.
+* Instance section, `instancesec`, declares and defines local instances.
+* Alias section, `aliassec`, brings in exports from nested instances or items
+  from the parent module into the local index space.
+
+The Type, Import, Module, Instance, and Alias sections can appear in any order
+at the beginning of a module. Each item defined in these sections can only refer
+to previously defined items. For example instance 4 cannot reference instance 5,
```

Comment:
I'm not personally familiar enough with how mutually-recursive types are expected to be used that it may be best if others make a PR to update the section in that regards. Is this for other proposals like gc types, or is it required for module linking itself? (I'm not sure if it'd be required for mutually recursive module types as-is with only this proposal myself)